### PR TITLE
[AAE-9019] - People/Group cloud with HxP

### PR DESCRIPTION
**Pull request by mauriziovitale**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/services/identity-group.service.ts [hunk 1]
```diff
@@ -276,6 +308,7 @@ export class IdentityGroupService { // implements IdentityGroupServiceInterface
         return this.getGroupRoles(groupId).pipe(map((groupRoles) => {
             let hasRole = false;
             if (groupRoles?.length > 0) {
+                //  fix the bug to include all  const roleResult = groupRoles.map( (role) =>  role.name).map ( (role) => roleNames.every(roleName => role.includes(roleName))
```
Comment (DenysVuika) [inline lib/core/services/identity-group.service.ts]: comments in the code

**Review by popovicsandras — APPROVED** [review]

File: demo-shell/src/app/app.routes.ts [hunk 2]
```diff
@@ -183,7 +183,7 @@ export const appRoutes: Routes = [
             {
                 path: 'cloud',
                 canActivate: [AuthGuardSsoRoleService],
-                data: { roles: ['ACTIVITI_ADMIN', 'ACTIVITI_USER'], redirectUrl: '/error/403' },
```
Comment (tomgny) [inline demo-shell/src/app/app.routes.ts:186]: we don't need the roles check anymore?

File: lib/process-services-cloud/src/lib/group/services/identity-group-filter.interface.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface IdentityGroupFilterInterface {
+    roles: string[];
```
Comment (tomgny) [inline lib/process-services-cloud/src/lib/group/services/identity-group-filter.interface.ts:19]: roles shoudn't be optional?
`roles?: string[];`

File: lib/process-services-cloud/src/lib/group/services/identity-group.service.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+    constructor(
+        private oAuth2Service: OAuth2Service,
+        private appConfigService: AppConfigService
+    ) {}
+
+    public search(name: string, filters?: IdentityGroupFilterInterface): Observable<IdentityGroupModel[]> {
+        if (name.trim() === '') {
+            return EMPTY;
+        } else if (filters?.withinApplication !== undefined && filters?.withinApplication !== '') {
```
Comment (tomgny) [inline lib/process-services-cloud/src/lib/group/services/identity-group.service.ts:40]: `filters?.withinApplication` shoudn't be enough? 
[ECMA docs](https://262.ecma-international.org/5.1/#sec-9.2)

File: lib/process-services-cloud/src/lib/group/services/identity-group.service.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+            return this.searchGroupsWithinApp(name, filters.withinApplication, filters.roles);
```
Comment (tomgny) [inline lib/process-services-cloud/src/lib/group/services/identity-group.service.ts:41]: i think we should check if roles are present
`filters?.roles`

File: angular.json [hunk 6]
```diff
@@ -843,8 +843,5 @@
       }
     }
   },
-  "defaultProject": "demoshell",
-  "cli": {
-    "analytics": false
-  }
-}
\ No newline at end of file
+  "defaultProject": "demoshell"
```
Comment (mauriziovitale) [inline angular.json:846]: why are we removing it?

Comment (tomgny) [reply angular.json:846]: @pmartinezga added it accidentally 5 days ago here: https://github.com/Alfresco/alfresco-ng2-components/pull/7652/files#diff-2d2675bb4687601a5c7ccf707455132f90f3516a33716185687e5c41df59ac7dR846-R850
We should remove that.

**Review by tomgny — APPROVED** [review]

**Pull request merged**